schema/yaml/shared.yml

```diff
@@ -29,6 +29,30 @@ message:
       type: object
       description: 'content of message'
 
+metadata:
+  id: 'metadata'
+  description: 'structure of key-value pairs for graph entity metadata'
+  type: object
+  additionalProperties: true
+  required: []
+  properties:
+    x:
+      type: integer
+      description: "X coordinate of a graph entity"
+    y:
+      type: integer
+      description: "Y coordinate of a graph entity"
+    route:
```

Comment:
Not all of these are valid for all kinds of metadata? Ex: `secure` and `route` is only on edges ?

Reply:
Correct. In flowbased/fbp#76 I did different metadata definitions for different things. We could go that granular here if we want to as well

Reply:
I think we should, it is a bit misleading right now

Reply:
OK, changed